src/umzug.ts

```diff
@@ -377,6 +370,18 @@ export class Umzug<Ctx extends object = object> extends emittery<UmzugEvents<Ctx
 				toWrite.push([filepath, '']);
 			}
 
+			if (!options.allowConfusingOrdering) {
+				const confusinglyOrdered = existing.find(e => {
+					const existingPath = e.path;
+					return existingPath && toWrite.some(([newPath]) => existingPath > newPath);
+				});
+				if (confusinglyOrdered) {
+					throw new Error(
+						`Can't create ${fileBasename}, since it's unclear if it should run before or after existing migration ${confusinglyOrdered.name}. Use allowConfusingOrdering to bypass this error.`
```

Comment:
There's another inconsistency. The error say "before or after", but the code checks that the new migration comes strictly after all existing migrations.

Reply:
Ah, makes sense. What if the message would explain it more or less how you just explained it? Or would it be to verbose?

`Can't create ${fileBasename}, since it's unclear if it should run before or after existing migration ${confusinglyOrdered.name}. Chronologically it's being created after ${confusinglyOrdered.name}, but alphabetically it comes before. Use allowConfusingOrdering to bypass this error.`